src/index.ts

```diff
@@ -213,21 +102,44 @@ export class OperationContext {
 	}
 
 	/**
-	 * Creates a new stack entry in the operation. Ideally, call this when calling a new
-	 * asynchronous operation to track its context separately while attached to the high-level
-	 * operation.
+	 * Sets one or multiple values on the current context. If the keys already
+	 * exist, they will be overwritten.
+	 * @param values additional values to append
 	 */
-	next(): OperationContextEntry {
+	setValues(values: Record<string, any>): OperationContext {
 		if (this.timeoutError) {
 			throw this.timeoutError
 		}
 		if (!this.isRunning()) {
-			throw this.createError(`Cannot continue a ${this.status} operation`)
+			throw this.createError(`Cannot set values on a ${this.status} operation`)
 		}
+		this.stack.push({ values, error: new Error('---') })
+		return this
+	}
 
-		const entry = new OperationContextEntry(this)
-		this.trace.push(entry)
-		return entry
+	/**
+	 * Given a request, appends the key information onto the current context.
+	 * @param request the http request
+	 * @param response the http response
+	 */
+	addHttpRequest(
+		request?: {
+			method: string
+			url: string
+			headers: Record<string, string>
```

Comment:
While we both think this should always be included, it may be an inconvenience to 3rd party open source users who aren't using a lib that outputs the headers it adds and haven't added any themselves, forcing them to write `{}`.  If you're okay with that, I am too - just something to consider.